File: easy/matrix/Modify the Matrix.py
class Solution(object):
    def modifiedMatrix(self, matrix):
        m = len(matrix)
        n = len(matrix[0])

        for i in range(n):
            l = -1000000
            for j in range(m):
                l = matrix[j][i] if matrix[j][i] > l else l
            
            for j in range(m):
                if matrix[j][i] == -1:
                    matrix[j][i] = l

        return matrix

File: easy/matrix/Modify the Matrix.cpp
class Solution {
public:
    vector<vector<int>> modifiedMatrix(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();
        
        for (int col = 0; col < n; ++col) {
            int max_col = INT_MIN;
            for (int row = 0; row < m; ++row) {
                if (matrix[row][col] != -1 && matrix[row][col] > max_col) {
                    max_col = matrix[row][col];
                }
            }
            
            for (int row = 0; row < m; ++row) {
                if (matrix[row][col] == -1) {
                    matrix[row][col] = max_col;
                }
            }
        }
        
        return matrix;
    }
};